shanoir-ng-preclinical/src/main/java/org/shanoir/ng/preclinical/migration/DistantPreclinicalShanoirService.java

```diff
@@ -0,0 +1,277 @@
+package org.shanoir.ng.preclinical.migration;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.shanoir.ng.preclinical.anesthetics.anesthetic.Anesthetic;
+import org.shanoir.ng.preclinical.anesthetics.examination_anesthetics.ExaminationAnesthetic;
+import org.shanoir.ng.preclinical.anesthetics.ingredients.AnestheticIngredient;
+import org.shanoir.ng.preclinical.pathologies.Pathology;
+import org.shanoir.ng.preclinical.pathologies.pathology_models.PathologyModel;
+import org.shanoir.ng.preclinical.pathologies.subject_pathologies.SubjectPathology;
+import org.shanoir.ng.preclinical.references.Reference;
+import org.shanoir.ng.preclinical.subjects.AnimalSubject;
+import org.shanoir.ng.preclinical.therapies.Therapy;
+import org.shanoir.ng.preclinical.therapies.subject_therapies.SubjectTherapy;
+import org.shanoir.ng.shared.exception.ShanoirException;
+import org.shanoir.ng.shared.migration.DistantKeycloakConfigurationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DistantPreclinicalShanoirService {
```

Comment:
Hi Jean-Come, similar comment: it would be amazing to bring out this API-CLIENT code into some external component, that all these remote calls will be available, e.g. from one client library. Just an idea and in line with my other comments on the other microservices. By the way, I like a lot the architecture, that each microservice has its own MigrationService that operates the migration. Very nice!